### PR TITLE
HVG-242: App Navigation, Contract Detail, Insurance Card

### PR DESCRIPTION
**Pull request by hedvigoscar**
### Checklist

- [ ] Functionality is covered by an integration test
- [X] Functionality is accessible in engineering mode

### CODE REVIEW

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt [hunk 1]
```diff
@@ -32,6 +35,7 @@ class ContractDetailActivity : BaseActivity(R.layout.contract_detail_activity) {
             e { "Programmer error: ID not provided to ${this.javaClass.name}" }
             return
         }
+        model.data.observe(this) { it.bindTo(binding.card) }
         model.loadContract(id)
```
Comment (MathiasOhrstrom) [inline app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt:39]: The arrow is still visible and the card is clickable here

**Review by MathiasOhrstrom — CHANGES_REQUESTED** [review]

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt (hunk 1 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt:39]: Good catch. 😬

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt [hunk 2]
```diff
@@ -61,7 +75,17 @@ class ContractDetailActivity : BaseActivity(R.layout.contract_detail_activity) {
...
             }.attach()
+            card.arrow.isVisible = false
+            card.root.transitionName = "contract_card"
         }
+
+        model.data.observe(this) {
+            it.bindTo(binding.card)
+            binding.card.apply {
+                arrow.isVisible = false
+            }
```
Comment (MathiasOhrstrom) [inline app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt:86]: Is this still needed in the observer?

**Review by MathiasOhrstrom — APPROVED** [review]

File: app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt (hunk 2 above)
Comment (hedvigoscar) [reply app/src/main/java/com/hedvig/app/feature/insurance/ui/detail/ContractDetailActivity.kt:86]: Good catch!

**Pull request merged**